File: src/main/java/com/vsu/by/app/education/attempt/Attempt.java
package com.vsu.by.app.education.attempt;

import com.vsu.by.app.education.pupilattempt.PupilAttempt;
import com.vsu.by.app.education.task.Task;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Data
@RequiredArgsConstructor
@Entity
public class Attempt {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id private Long id;
    @OneToMany(mappedBy = "attempt", cascade = {CascadeType.MERGE})
    private List<PupilAttempt> pupilsAttempts;
    @OneToOne
    private Task task;
    private Long start;
    private Long end;
}

File: angular/testing-programm/src/app/attempt/attempt.ts
import { Task } from "../task/task";
import { PupilAttempt } from "../pupil-attempt/pupilAttempt";

export class Attempt {
    id: number;
    pupilsAttempts: PupilAttempt[] = new Array<PupilAttempt>();
    task: Task;
    start: number;
    end: number;
}